File: js/breadthFirstSearch.js
const graph = {};

// FIRST CHAIN LINK. You are Adam and you search a seller
graph["Adam"] = ["Avram", "Bollet"];

// CHAIN FROM ADAM
graph["Avram"] = ["Sandra", "Late Seller"];
graph["Bollet"] = ["Valera Seller"];

graph["Sandra"] = [];
graph["Late Seller"] = [];
graph["Valera Seller"] = [];

const getIsSeller = (name) => (name.includes("Seller") ? true : false);

const searchOfSeller = (name) => {
  const searchQueue = [];
  searchQueue.push(...graph[name]);
  const searchedPerson = [];

  while (searchQueue.length > 0) {
    const person = searchQueue.shift();

    if (!searchedPerson.includes(person)) {
      if (getIsSeller(person)) {
        console.log(`i got it. ${person} is a seller!`);
        return true;
      } else if (graph[person]) {
        searchQueue.push(...graph[person]);
        searchedPerson.push(person);
        console.log(`${person} is not a seller!`);
      } else continue;
    }
  }
  return false;
};

console.log(searchOfSeller("Adam"));

File: python/breadth_first_search.py
import collections

graph = {}
# FIRST CHAIN LINK. You are Adam and you search a seller
graph["Adam"] = ["Avram",  "Bollet"]

# CHAIN FROM ADAM
graph["Avram"] = ["Sandra", "Late Seller"]
graph["Bollet"] = ["Valera Seller"]

graph["Sandra"] = []
graph["Late Seller"] = []
graph["Valera Seller"] = []


def is_seller(name):
    return "Seller" in name


def search_of_seller(name):
    search_queue = collections.deque()
    search_queue += graph[name]
    searched_persons = []
    while search_queue:
        person = search_queue.popleft()
        if not person in searched_persons:
            if is_seller(person):
                print(f"I got it. {person} is a seller!")
                return True
            else:
                if graph[person]:
                    search_queue += graph[person]
                    print(f'{person} not a seller!')
                    searched_persons.append(person)
                else:
                    continue

    return False


search_of_seller("Adam")
